### PR TITLE
feat(devtools): Add accessibilty test coverage for Devtools

### PR DESCRIPTION
**Pull request by WayneFerrao**
[AB#8206](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8206)

## Description

This PR adds  accessibility test coverage for Fluid Dev Tools to the following components:
- Menu
- AudienceStateTable
- AudienceHistoryTable 
- OpLatencyView

### CODE REVIEW

File: packages/tools/devtools/devtools-view/src/DevtoolsView.tsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ interface OpLatencyMenuSelection {
  * E.g. {@link ContainerMenuSelection} represents that the menu option for a Container
  * is selected, and has a 'containerKey' property to indicate which Container.
  */
-type MenuSelection =
+export type MenuSelection =
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/DevtoolsView.tsx:117]: This whole type union is duplicated here and in src/components/Menu.tsx, I think only the latter should exist. Can we clean it up here too?

File: packages/tools/devtools/devtools-view/src/test/AudienceHistoryTable.test.tsx [hunk 2]
```diff
@@ -55,3 +58,18 @@ describe("AudienceHistoryTable component tests", () => {
...
 	});
 });
+
+describe("AudienceHistoryTable Accessibility Check", () => {
+	it("AudienceHistoryTable is accessible", async () => {
+		const { container } = render(<AudienceHistoryTable audienceHistoryItems={[]} />);
+		await assertNoAccessibilityViolations(container);
+	});
+
+	it("Can tab/arrow navigate throught AudienceHistoryTable", async () => {
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/AudienceHistoryTable.test.tsx:68]: Nit
```suggestion
	it("Can tab/arrow navigate through AudienceHistoryTable", async () => {
```

File: packages/tools/devtools/devtools-view/src/test/Menu.test.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+				setSelection={setMenuSelection}
+				containers={containers}
+				supportedFeatures={supportedFeatures}
+			/>
+		);
+	};
+	const mockMessageRelay = new MockMessageRelay(() => {
+		return {
+			type: DevtoolsFeatures.MessageType,
+			source: "OpLatencyTest",
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/Menu.test.tsx:44]: At runtime this has actually meaningful values. In tests when we're mocking most things I think it won't matter most of the time, but let's keep it in sync with the test that's using this mock relay.
```suggestion
			source: "MenuAccessibilityTest",
```

File: packages/tools/devtools/devtools-view/src/test/Menu.test.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			data: {
+				features: {
+					telemetry: true,
+					opLatencyTelemetry: true,
+				},
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/Menu.test.tsx:49]: Should this reuse the const from line 23?

File: packages/tools/devtools/devtools-view/src/test/Menu.test.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+				unsampledTelemetry: true,
+			},
+		};
+	});
+
+	it("Menu is accessible", async () => {
+		const { container } = render(
+			<MessageRelayContext.Provider value={mockMessageRelay}>
+				<MenuWrapper />
+			</MessageRelayContext.Provider>,
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/Menu.test.tsx:60]: For test purposes I think the relay can also go inside the wrapper. We'd expect every test in this file to have to do this same thing, right?

Comment (WayneFerrao) [reply packages/tools/devtools/devtools-view/src/test/Menu.test.tsx:60]: Yeah good point

File: packages/tools/devtools/devtools-view/src/test/AudienceStateTable.test.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	it("AudienceStateTable is accessible", async () => {
+		const { container } = render(<AudienceStateTable audienceStateItems={[]} />);
+		await assertNoAccessibilityViolations(container);
+	});
+
+	it("Can tab/arrow navigate through the AudienceStateTable", async () => {
+		render(<AudienceStateTable audienceStateItems={[]} />);
+
+		const user = userEvent.setup();
+		// Focus on the first interactive element (dropdown theme selector)
```
Comment (jikim-msft) [inline packages/tools/devtools/devtools-view/src/test/AudienceStateTable.test.tsx:27]: I think this comment applies to the `SettingsView.test.tsx`, so we should probably remove it.

**Comment by jikim-msft** [comment]
Left one nit comment, but everything LGTM.

Are we planning to have a follow-up work on testing `ContainerSummaryView`? I think it will be really nice if we can test how components like `ContainerDataView`, `AudienceStateTable` & `AudienceHistoryTable` work differently based on the state of the `ContainerSummaryView`. 

Screenshot 2024-06-28 at 17 14 47


For example, once the container is closed. The user should have focus on `Client ID` of `AudienceHistoryTable` after `User ID` of `ContainerSummaryView`.

**Review by jikim-msft — APPROVED** [review]

**Review by alexvy86 — APPROVED** [review]
Agree with Ji's comment above and the nit to remove a code comment that applies to the test for settings view but not audience state table. Overall lgtm!

**Comment by WayneFerrao** [comment]
@jikim-msft @alexvy86 Yes, there is a task tracking the remaining components that need coverage and I updated it to include ContainerSummaryView

**Pull request merged**